zebra/zebra_rib.c

```diff
@@ -2412,7 +2424,13 @@ void rib_unlink(struct route_node *rn, struct route_entry *re)
 	if (dest->selected_fib == re)
 		dest->selected_fib = NULL;
 
-	nexthops_free(re->ng.nexthop);
+	if (re->nhe_id) {
+		nhe = zebra_nhg_lookup_id(re->nhe_id);
+		if (nhe)
+			zebra_nhg_decrement_ref(nhe);
+	} else if (re->ng)
+		nexthop_group_delete(&re->ng);
+
 	nexthops_free(re->fib_ng.nexthop);
```

Comment:
are we not using the 'fib nexthop-group' concept? we did add that to try to deal with a couple of problems

Reply:
That is still retained, its just not created as a nhg_hash_entry.

When we enable the kernel nexthop group API changes will need to be made there though. Still figuring that one out. They might not even be useful at that point to be honest.